default/scripting/species/common/focus.macros

```diff
@@ -10,15 +10,37 @@ HAS_RESEARCH_FOCUS
 '''        Focus
             name = "FOCUS_RESEARCH"
             description = "FOCUS_RESEARCH_DESC"
-            location = Planet
+            location = And [
+                Planet
+                Or [
+                    Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                    Number low = 1 condition = And [
+                        Planet
+                        OwnedBy empire = Source.Owner
+                        Species name = Source.Species
+                        Contains Building name = "BLD_IMPERIAL_PALACE"
+                    ]
+                ]
+            ]
             graphic = "icons/focus/research.png"
 '''
 
 HAS_INFLUENCE_FOCUS
 '''        Focus
             name = "FOCUS_INFLUENCE"
             description = "FOCUS_INFLUENCE_DESC"
-            location = Planet
+            location = And [
+                Planet
+                Or [
+                    Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                    Number low = 1 condition = And [
+                        Planet
+                        OwnedBy empire = Source.Owner
+                        Species name = Source.Species
+                        Contains Building name = "BLD_IMPERIAL_PALACE"
```

Comment:
Checking for using Capital condition as I and geoff hinted is probably cleaner than checking for the building. Here you could simply throw away the Contains.. and use Capital instead (if i am right)

Reply:
I know, I did that a couple of days ago.